tools/urinfo/utils.hpp

```diff
@@ -0,0 +1,205 @@
+// Copyright (C) 2023 Intel Corporation
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+// See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+
+#include "ur_api.h"
+#include "ur_params.hpp"
+#include <cstdlib>
+#include <iostream>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <vector>
+
+#define UR_CHECK(ACTION)                                                       \
+    if (auto error = ACTION) {                                                 \
+        std::cerr << "error: " #ACTION " failed: " << error << "\n";           \
+        std::exit(1);                                                          \
+    }                                                                          \
+    (void)0
+
+#define UR_CHECK_WEAK(ACTION)                                                  \
+    if (auto error = ACTION) {                                                 \
+        std::cout << error << "\n";                                            \
+        return;                                                                \
+    }                                                                          \
+    (void)0
+
+namespace urinfo {
+// TODO: Generate this
```

Comment:
Not sure if that's needed. There's a function to print the enum (https://github.com/oneapi-src/unified-runtime/blob/main/source/common/ur_params.hpp#L2153), from that you can just remove the prefix and `tolower` the result.
Same for `ur_device_type_t`.

Reply:
Good shout.